parallel-install/pkg/deployment/coredns_test.go

```diff
@@ -5,50 +5,83 @@ import (
 
 	"github.com/kyma-incubator/hydroform/parallel-install/pkg/logger"
 	"github.com/stretchr/testify/require"
+	v1apps "k8s.io/api/apps/v1"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes/fake"
 )
 
 func Test_patchCoreDNS(t *testing.T) {
 	log := logger.NewLogger(true)
-	coreDNSCM := fakeCoreDNSCM()
+	coreDNSConfigMap := fakeCoreDNSConfigMap()
+	coreDNSDeployment := fakeCoreDNSDeployment()
 
-	t.Run("test skipping coreDNS patch", func(t *testing.T) {
+	t.Run("test skipping coreDNS patch when coreDNS deployment doesn't exist", func(t *testing.T) {
 		// given
-		kubeClient := fake.NewSimpleClientset(coreDNSCM)
+		kubeClient := fake.NewSimpleClientset(coreDNSConfigMap)
 
 		// when
-		cm, err := patchCoreDNS(kubeClient, `(.*)\.loacal\.kyma\.dev`, log)
+		cm, err := patchCoreDNS(kubeClient, `(.*)\.local\.kyma\.dev`, log)
 
 		// then
 		require.NoError(t, err)
 		require.Empty(t, cm.Data)
 	})
+
+	t.Run("test skipping coreDNS patch when coreDNS configMap has proper entry", func(t *testing.T) {
+		// given
+		kubeClient := fake.NewSimpleClientset(coreDNSConfigMap, coreDNSDeployment)
+
+		// when
+		cm, err := patchCoreDNS(kubeClient, `(.*)\.local\.kyma\.dev`, log)
+
+		// then
+		require.NoError(t, err)
+		require.Empty(t, cm.Data)
+	})
+
+	t.Run("test patching coreDNS configMap when coreDNS configMap is empty", func(t *testing.T) {
+		// given
+		emptyConfigMap := &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "coredns",
+				Namespace: "kube-system",
+			},
+			Data: make(map[string]string),
+		}
+		kubeClient := fake.NewSimpleClientset(coreDNSDeployment, emptyConfigMap)
+		domain := `(.*)\.local\.kyma\.dev`
+
+		// when
+		cm, err := patchCoreDNS(kubeClient, domain, log)
+
+		// then
+		require.NoError(t, err)
+		require.Contains(t, cm.Data["Corefile"], domain)
+	})
+
+	t.Run("test patching coreDNS configMap when coreDNS configMap is wrong", func(t *testing.T) {
```

Comment:
What is wrong with the domain? Can you rename the test-case to be more self-descriptive?